### PR TITLE
[big.js] Fix export kind

### PR DESCRIPTION
**Pull request by andrewbranch**
The primary purpose of this PR is to replace the incorrect `export default` with an `export =`. This was brought to my attention at https://github.com/microsoft/TypeScript/issues/50058#issuecomment-1644335391.

Doing this requires a bit of refactor to merge everything exported into a single symbol named `Big`. Additionally, the only way to make the necessary merges legal is to eliminate the deprecated `const enum` declarations. These were deprecated in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/58523, so I think it has been long enough that we can remove them, in a tradeoff that allows these types to be used in ESM files in `--module nodenext`. There has been a new minor version (6.2) released since last time the header version number was updated, so I bumped the header version to minimize the chances that users will be impacted without explicitly installing a new version.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://unpkg.com/big.js@6.2.1/big.js
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/big.js/tsconfig.json
Comment (andrewbranch) [inline types/big.js/tsconfig.json:18]: Normally, we prefer not to use `esModuleInterop` in DT configs, because it can allow non-portable import statements (ones that will error for consumers unless they too set `esModuleInterop`) to appear in the types themselves. However, these definitions have no imports in the types, and the tests do a good job of checking different ways of importing the module, in ways that should require `esModuleInterop` to be set. Setting these here allowed me to keep the imports in the tests as-is, which ensures that the module-related changes are not breaking for users as long as they have `esModuleInterop` set.

**Comment by jstasiak** [comment]
Works great, just tested this locally.

File: types/big.js/index.d.ts
Comment (rbuckton) [inline types/big.js/index.d.ts:355]: Isn't the normal way to do this to just write a merged declaration? i.e.:

```ts
declare function Big(): typeof Big;
declare function Big(value: Big.BigSource): Big;
declare class Big {
  constructor(value: Big.Bigsource);
  // either declare statics for DP/RM/etc. or make them namespace exports?
}
interface Big {
 ... // maybe just merge into class above? 
}
declare namespace Big {
  // everything else in Big namespace
}
export = Big;
```

We will merge the `function` and `class` declarations in the .d.ts file, so there's no need to declare a `BigConstructor` (though you could do `export type BigConstructor = typeof Big` if you wanted to)

Comment (andrewbranch) [reply types/big.js/index.d.ts:355]: Yeah, I’m not sure why this was initially declared as a separate constructor. I was trying not to change stuff that I didn’t have to change as part of this refactor, but maybe that’s worthwhile.

Comment (andrewbranch) [reply types/big.js/index.d.ts:355]: Oh, I think I see a reason. If you call (not construct) the “constructor” function with no arguments, it returns _another constructor function_ https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66163/files#diff-8034f96ce0019a4b4f9f3431da74d25feaa41073c0f68ca5f02f4356eb3076ffR54

I don’t think this is something a class declaration can support.

Comment (rbuckton) [reply types/big.js/index.d.ts:355]: That's why the example I posted above has both `declare function Big()` and `declare class Big`. We allow function/class merging when both declarations are ambient types.

Comment (andrewbranch) [reply types/big.js/index.d.ts:355]: Oof, TIL.

Is there any reason to make that refactor now? It’s been like this for a long time.

Comment (rbuckton) [reply types/big.js/index.d.ts:355]: That's fine. I just figured if we're making a change we might as well use a cleaner style, but its not an issue.

**Review by rbuckton — APPROVED** [review]

**Comment by andrewbranch** [comment]
It would be great to get a sign-off from @nycdotnet on the removal of the const enums, which were deprecated in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/58523 over a year ago. I’ll wait until next week to merge.

**Comment by andrewbranch** [comment]
Ready to merge

**Comment by jstasiak** [comment]
I just discovered a regression that appears to be related to this change, cross-referencing a TS report: https://github.com/microsoft/TypeScript/issues/56673

**Pull request merged**
